### PR TITLE
Documentation - Docusaurus port

### PR DESCRIPTION
**Pull request by InfraredAces**
This port should contain all of the current documentation that is live on the [documentation site](https://gp2040-ce.info/#/) as of the time of this PR.

I have a test version of this site running one https://infraredaces.github.io/docusaurus-test/ , if you would like to see it live and test.

There are only a few things left to do before and after this PR is merged to allow the workflow to deploy the site, but they are only performable by maintainers of the repo.

- Before: Create an access token with `Read and Write` access to the Contents of the repo, name it "ACTION_SECRET", and then place it in Repository Secrets. The name can be changed, however the deploy.yml contents will need to be updated as well.

image

- After: Change the target of the Github Pages to the generated `gh-pages` branch that the `deploy` action creates. 

image

This action will create a branch of that consists of only the static files generated during the build process and result in the pages seen in the test site linked above.

image

### CODE REVIEW

**Review by arntsonl — COMMENTED** [review]
Going to wait for at least one other review from a web-design stand-point, but I commented on things that stood out to me.

File: site/docs/development/firmware-development.mdx [hunk 1]
```diff
@@ -36,10 +43,8 @@ For GP2040-CE development on the Windows platform using the Pico SDK Installer,
 9. Install "CMake Tools" by Microsoft.
 10. Open the GP2040-CE folder you downloaded earlier via "File > Open Folder."
 
-### **Windows (Manual)**
-
-### Windows (Manual)
-
+</TabItem>
```
[the remark this answers is not in the document]
Comment (InfraredAces) [reply site/docs/development/firmware-development.mdx:46]: Having the Tabs and TabItems properly indented results in an error due to the lack of closing tag, which are clearly present in the document. By having all the Tags unindented, the issue is resolved. Unsure of how to proceed.

image

File: site/.gitignore [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+packages/docusaurus-*/lib/*
+packages/eslint-plugin/lib/
+packages/stylelint-copyright/lib/
+
+website/netlifyDeployPreview/*
+website/changelog
+!website/netlifyDeployPreview/index.html
+!website/netlifyDeployPreview/_redirects
+
+website/_dogfooding/_swizzle_theme_tests
```
[the remark this answers is not in the document]
Comment (InfraredAces) [reply site/.gitignore:37]: These are not specific to my setup, this is the .gitignore used by Docusaurus. `website` is the default name of the folder when generating a site using a template. Will remove unnecessary references.

https://github.com/facebook/docusaurus/blob/main/.gitignore

File: site/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Website
```
[the remark this answers is not in the document]
Comment (InfraredAces) [reply site/README.md:1]: Removed Docusaurus README

File: site/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+				logo: {
+					alt: 'GP2040-CE Logo',
+					src: 'img/gp2040-ce-logo.svg',
+				},
+				items: [
+					{
+						type: 'docSidebar',
+						position: 'left',
+						sidebarId: 'docSidebar',
+						label: 'Docs',
```
[the remark this answers is not in the document]
Comment (InfraredAces) [reply site/docusaurus.config.js:96]: Changed from "Docs" to "Home"

File: site/docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+			},
+			announcementBar: {
+				id: 'new_release',
+				content:
+					`New Version Available! To get the v${releaseVersion} update, go to <a href="https://github.com/OpenStickCommunity/GP2040-CE/releases/tag/v${releaseVersion}" target="_blank">GP2040-CE Releases<a/>`,
+				backgroundColor: '#ec008c',
+				textColor: '#FFFFFF',
+				isCloseable: true,
+			},
+			// Replace with your project's social card
```
[the remark this answers is not in the document]
Comment (InfraredAces) [reply site/docusaurus.config.js:83]: Removed built in comments from Docusaurus Template generation.

File: site/docs/development/firmware-development.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (InfraredAces) [reply site/docs/development/firmware-development.mdx:46]: This seems to be an issue with interoperability between Markdown and JSX.

https://docusaurus.io/docs/markdown-features/react#markdown-and-jsx-interoperability

File: README.md [hunk 6]
```diff
@@ -14,10 +15,16 @@
...
+  <img src="https://img.shields.io/github/license/OpenStickCommunity/GP2040-CE" />
+  <img src="https://img.shields.io/github/actions/workflow/status/OpenStickCommunity/GP2040-CE/cmake.yml" />
+  <br />
+  <img src="https://img.shields.io/badge/inputlag.science-0.86%20ms-blue" />
+  <img src="https://img.shields.io/badge/MiSTer%20latency-0.765%20ms-blue" />
 </p>
 
 <p>
   GP2040-CE (Community Edition) is a gamepad firmware for the Raspberry Pi Pico and other boards based on the RP2040 microcontrollers that combines multi-platform compatibility, low latency and a rich feature set to provide endless customization possibilities without sacrificing performance.
+  GP2040-CE (Community Edition) is a gamepad firmware for the Raspberry Pi Pico and other boards based on the RP2040 microcontrollers that combines multi-platform compatibility, low latency and a rich feature set to provide endless customization possibilities without sacrificing performance.
```
[the remark this answers is not in the document]
Comment (InfraredAces) [reply README.md:27]: Indeed, will remove

File: site/docs/development/firmware-development.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (arntsonl) [reply site/docs/development/firmware-development.mdx:46]: Ahhh okay, yeah so a technical limitation! No worries

**Review by arntsonl — COMMENTED** [review]
Nice! I'll wait for another reviewer until we merge but changes look good!

File: configs/Pico/BoardConfig.h [hunk 7]
```diff
@@ -317,7 +317,7 @@
 #define PSPASSTHROUGH_HOST_PIN_DPLUS -1
 #define PSPASSTHROUGH_HOST_PIN_5V -1
 
-// For details on this, see: https://gp2040-ce.info/#/development?id=i2c-display-splash
+// For details on this, see: /https://gp2040-ce.info/development?id=i2c-display-splash
```
Comment (arntsonl) [inline configs/Pico/BoardConfig.h:320]: Looks like a bad '/' got merged in here

File: site/docs/assets/boards/ReflexEncoder.jpg
Comment (arntsonl) [inline site/docs/assets/boards/ReflexEncoder.jpg:1]: Did we replace this with V1/V2?

**Review by arntsonl — APPROVED** [review]
Looks really good!

**Comment by Pelsin** [comment]
Looks great!
Awesome work!

File: site/docs/assets/boards/ReflexEncoder.jpg
Comment (InfraredAces) [reply site/docs/assets/boards/ReflexEncoder.jpg:1]: This board has been replaced with ReflexEncode_v1.2.jpg and ReflexEncode_v2.0.jpg so we should remove

**Pull request merged**